src/auth/src/credentials.rs

```diff
@@ -165,3 +165,80 @@ pub mod traits {
         }
     }
 }
+
+/// The path to Application Default Credentials (ADC), as specified in [aip/4110].
+///
+/// [aip/4110]: https://google.aip.dev/auth/4110
+#[allow(dead_code)] // TODO(#442) - implementation in progress
+fn adc_path() -> String {
+    if let Ok(e) = std::env::var("GOOGLE_APPLICATION_CREDENTIALS") {
+        return e;
+    }
+    adc_well_known_path()
+}
+
+/// The well-known path to ADC on Windows, as specified in [aip/4113].
+///
+/// [aip/4113]: https://google.aip.dev/auth/4113
+#[cfg(target_os = "windows")]
+fn adc_well_known_path() -> String {
+    let root = std::env::var("APPDATA").unwrap();
```

Comment:
This will panic and crash the application.  I do not believe that is the behavior we want.  It should be something like "there is no ADC file".

Reply:
Yeah, good catch. I assumed I would have a `HOME` / `APPDATA`.

Fixed by propagating errors. I kind of hate the error type though. I used `CredentialError`, but I don't think I want that. I think I want a new type that makes my life easier in the credential factory functions... Let me think about it.

Reply:
`Option<T>` may do the trick.  It is made for "I wanted a T but it is not set / not found".

Reply:
And btw, `HOME` is often missing when running inside a docker container.

Reply:
> `Option<T>` may do the trick.

Yeah, thanks, I prefer this.

If we don't have ADC it is not an error, we are supposed to fallback to MDS credentials. So returning an error here did not make much sense.